test/api_server_unittest.cc

```diff
@@ -0,0 +1,78 @@
+#include "../src/api_server.h"
+#include "../src/configuration.h"
+#include "../src/store.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class ApiServerTest : public ::testing::Test {
+ protected:
+  using Dispatcher = MetadataApiServer::Dispatcher;
+  std::unique_ptr<Dispatcher> CreateDispatcher(
+      const std::map<std::pair<std::string, std::string>,
+                     std::function<void()>>& handlers) {
+    Dispatcher::HandlerMap handler_map;
+    for (const auto& element : handlers) {
+      std::function<void()> handler = element.second;
+      handler_map.emplace(element.first, [handler](
+          const MetadataApiServer::HttpServer::request& request,
+          std::shared_ptr<MetadataApiServer::HttpServer::connection> conn) {
+        handler();
+      });
+    }
+    return std::unique_ptr<Dispatcher>(new Dispatcher(handler_map, false));
+  }
+
+  void InvokeDispatcher(
+      const std:: unique_ptr<Dispatcher>& dispatcher,
+      const std::string& method, const std::string& path) {
+    MetadataApiServer::HttpServer::request request;
+    request.method = method;
+    request.destination = path;
+    (*dispatcher)(request, nullptr);
+  }
+};
+
+TEST_F(ApiServerTest, BasicDispacher) {
+  bool handler_called = false;
+  std::unique_ptr<Dispatcher> dispatcher = CreateDispatcher({
+    {{"GET", "/testPath/"}, [&handler_called]() {
+      handler_called = true;
+    }},
+    {{"GET", "/badPath/"}, [&handler_called]() {
+      handler_called = false;
```

Comment:
There's not enough signal here (e.g., imagine if the implementation accidentally called every handler — if this one got called first, you'd never know). How about adding a `bad_handler_called` boolean for this one (and setting it to `true` in the handler)?